### ISSUE TITLE
RET504 - false positive

### ISSUE THREAD

**Issue by Czaki** [charliermarsh/ruff#1517]
I have checked that same problem also occurs in `flake8-return` so it is not a bug in logic translation, but definitely there is a problem with this rule. I also see #1035 with similar problems related to this plugin/set of rules.  

The following code is reported as breaking RET504.

```
    def get_changes(self):
        ret = self.changes
        self.changes = []
        return ret
```

On one side, it fits in the description, on the other side, I do not see any better way to write such code. 
Similarly to #1035 I do not use `flake8-return` earlier but would like to try these rules when seeing the description in the ruff readme.

**Comment by charliermarsh** [comment]
That makes sense. We could cause assignments to invalidate the rule. I think we already do that for function calls.
